src/components/forms/Buses.js

```diff
@@ -22,9 +25,29 @@ const Buses = ({
   });
   const [loading, setloading] = useState(false);
   const onValid = async (data) => {
-    setloading(true);
-    await formAction(data);
-    setloading(false);
+    try {
+      setloading(true);
+      await formAction(data);
+      setloading(false);
+      navigate(-1);
+    } catch (error) {
+      const validationErrors = error?.response?.data.data;
+      if (validationErrors) {
```

Comment:
Avoid nesting  of conditional operators and also you can move this to a custom hook to increase readability and reduce complexity